combat-trainer.lic

```diff
@@ -1876,6 +1881,40 @@ class TrainerProcess
     DRRoom.dead_npcs.size > dead_count
   end
 
+  def ambush_choke(game_state)
+    return false unless game_state.can_ambush_stun?
+    return false unless @dirt_stacker
+    return false if @dirt_empty
+
+    unless bput("push my #{@dirt_stacker}",
+                'You push', 'have any dirt', "I don't think pushing") == 'You push'
+      @dirt_empty = true
+      return false
+    end
+
+    dirt_in_hand = true
+    if hide?
+      dirt_in_hand = bput(
+        'ambush choke',
+        'Wouldn\'t it be better if you used a melee weapon?',
+        'You must be hidden or invisible to ambush',
+        'You aren\'t close enough to attack',
+        'You don\'t have enough focus',
+        'You don\'t think you have enough focus',
+        'Snipe dirt at somebody',
+        'Roundtime'
+      ) != 'Roundtime'
+      pause
+      waitrt?
+    end
+    if dirt_in_hand
+      bput("put my dirt in my #{@dirt_stacker}",
+           'dumping some dirt', 'What were you referring')
+    end
+
+    false
```

Comment:
I realize you're following the precedent of the other training abilities, but I'm not convinced the return value is used anywhere. What do you think?

Reply:
You'd need a return up in the case that calls this like `ambush_stun` has, but that's only necessary if the action here could cause a real change in game state, I don't think choke can kill mobs so there's no need.

Reply:
You're right, totally not needed.  Removed.